src/ZoneServer/Buffs/Handlers/Agility_Buff.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Melia.Shared.Tos.Const;
+using Melia.Zone.Buffs.Base;
+using Melia.Zone.Network;
+using Melia.Zone.Scripting;
+using Melia.Zone.Skills;
+using Melia.Zone.World.Actors.Characters;
+using Melia.Zone.World.Actors.Characters.Components;
+using Melia.Zone.World.Items;
+
+namespace Melia.Zone.Buffs.Handlers
+{
+	/// <summary>
+	/// Handle for the Agility Buff, which increases the target's movement speed.
+	/// </summary>
+	[BuffHandler(BuffId.Agility_Buff)]
+	public class Agility_Buff : BuffHandler
+	{
+		private const string VarNameMVSD = "Melia.MovementSpeedBonus";
+		private const string VarNameSTA = "Melia.StaminaRateBonus";
+
+		public override void OnStart(Buff buff)
+		{
+			var target = buff.Target as Character;
+
+			// It is not applyed to characters without Shoes
+			if (target != null && !(target.Inventory.GetEquip(EquipSlot.Shoes) is DummyEquipItem))
+			{
+				var skillLevel = buff.NumArg1;
+				var caster = buff.Caster as Character;
+
+				// Algorithm retrieved from client files.
+				var speedBonus = 1 + (skillLevel * 0.3);
+
+				if (caster.Components.Get<AbilityComponent>().Has(AbilityId.Enchanter10))
+				{
+					var SCR_Get_SkillFactor = ScriptableFunctions.Skill.Get("SCR_Get_SkillFactor");
```

Comment:
You should prefer using properties over explicitly calling scriptable functions. It's simpler and you can be sure that you actually get the property for this skill and not some partial value.